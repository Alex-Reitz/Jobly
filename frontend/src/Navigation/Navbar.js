import React, { useContext } from "react";
import "./Navbar.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";
import UserContext from "../Context/userContext";

function NavBar() {
  const user = useContext(UserContext);
  if (!user.username) {
    return (
      <div>
        <Navbar expand="md">
          <NavLink exact to="/" className="navbar-brand">
            Home
          </NavLink>
          <Nav className="ml-auto" navbar>
            <NavItem className="ml-auto">
              <NavLink to="/companies">Companies</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/jobs">Jobs</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/signup">Signup</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/login">Login</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/profile">Profile</NavLink>
            </NavItem>
          </Nav>
        </Navbar>
      </div>
    );
  } else {
    return (
      <div>
        <Navbar expand="md">
          <NavLink exact to="/" className="navbar-brand">
            Home
          </NavLink>
          <Nav className="ml-auto" navbar>
            <NavItem className="ml-auto">
              <NavLink to="/companies">Companies</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/jobs">Jobs</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/profile">Profile</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/logout">Logout</NavLink>
            </NavItem>
          </Nav>
        </Navbar>
      </div>
    );
  }
}

export default NavBar;
