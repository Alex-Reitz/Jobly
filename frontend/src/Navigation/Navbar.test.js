import React from "react";
import { render } from "@testing-library/react";
import Navbar from "./Navbar";
import { MemoryRouter } from "react-router";
import UserContext from "../Context/userContext";

it("matches snapshot", function () {
  const testFunc = jest.fn();
  const { asFragment } = render(
    <MemoryRouter>
      <UserContext.Provider value={{ currentUser: testFunc }}>
        <Navbar />
      </UserContext.Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
