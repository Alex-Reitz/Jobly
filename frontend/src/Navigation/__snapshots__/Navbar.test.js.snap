// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
  >
    <div
      class="nav-container"
    >
      <ul
        class="nav-menu"
      >
        <li
          class="nav-item"
        >
          <a
            aria-current="page"
            class="nav-links active"
            href="/"
          >
            Home
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-links"
            href="/login"
          >
            Login
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="nav-links"
            href="/signup"
          >
            Signup
          </a>
        </li>
      </ul>
      <div
        class="nav-icon"
      >
        <i
          class="fas fa-bars"
        />
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
